### ISSUE TITLE
Exported symbol PL_version conflicts with symbol in Perl

### ISSUE THREAD

**Issue by zmughal** [SWI-Prolog/swipl-devel#900]
The conflicting symbol `PL_version` prevents initialising SWI-Prolog at runtime from a library I maintain that uses both bidirectionally.

```sh
$ for e in /usr/bin/perl /lib/libswipl.so; do strings $e  | grep -H --label=$e ^PL_version$; done
/usr/bin/perl:PL_version
/lib/libswipl.so:PL_version
```

See <https://github.com/salva/p5-Language-Prolog-Yaswi/issues/3>.

**Comment by JanWielemaker** [comment]
I see.   This is a bit unfortunate.  Had I know about it I could have used e.g. PL_version_info() or something similar.   After a stable release it gets harder.    The best solution is probably to rename the function and provide a macro for backward compatibility.    There are several language bindings that directly access the shared object functions though, and these will break :cry:    The contract is that such changes are not made during a stable cycle.

On ELF (seems to be the case considering .so), this should not matter and you merely only have access to the first version of the symbol loaded while you can use dlsym() to get the specify one.   Something similar works for other systems AFAIK.

So, if there is a reasonable work around for you using the linker, I guess this is best.    If not, I'll consider the above rename and macro for the development series.

**Comment by zmughal** [comment]
Yes, I totally understand that this ABI change can not be done during the current development cycle.

Because of how the binding library is loaded and the Perl symbol being a variable needed for loading that library, using `dlsym()` would not work easily. Also, I'm trying to make sure this continues to be cross-platform (it was working on Windows as far back as SWI-Prolog 5.x).

For now my workaround is that I'm automatically building my own patched copy of SWI-Prolog if the system version won't build.

**Comment by JanWielemaker** [comment]
Thanks for your consideration.   As it is good to have this resolved somewhere in the future, I propose to rename PL_version into PL_version_info() with a macro that allows calling the old in de development version (stable remains as is).  Would that work for you?

**Comment by zmughal** [comment]
Yes, that works.

And thank you for SWI-Prolog!

**Comment by JanWielemaker** [comment]
Resolved with SWI-Prolog/swipl-devel@bf0fb835c075926931d67969ab524ade01197c9a
